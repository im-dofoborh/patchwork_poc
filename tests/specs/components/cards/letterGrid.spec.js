import { mount, createLocalVue } from '@vue/test-utils';
import LetterGridCard from '~/components/cards/LetterGrid';

describe('LetterGridCard component', () => {
  let localVue = null;
  let wrapper = null;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.component('letter-grid-card', LetterGridCard);
    wrapper = mount(LetterGridCard, {
      localVue,
    });
  })

  it('renders default variant', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders exploded variant', () => {
    wrapper.setProps({ explode: true });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders with alternative word', () => {
    wrapper.setProps({
      word: 'Foobar'
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders with alternative size', () => {
    wrapper.setProps({
      size: 400,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
