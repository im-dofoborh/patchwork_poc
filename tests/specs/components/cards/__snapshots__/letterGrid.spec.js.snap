// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LetterGridCard component renders default variant 1`] = `
<div class="sg-letter-grid-card">
  <div title="P" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 1;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      P
    </div>
  </div>
  <div title="a" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 0.9111111111111111;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      A
    </div>
  </div>
  <div title="t" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 0.8222222222222222;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      T
    </div>
  </div>
  <div title="c" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 0.7333333333333334;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      C
    </div>
  </div>
  <div title="h" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 0.6444444444444444;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      H
    </div>
  </div>
  <div title="w" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 0.5555555555555556;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      W
    </div>
  </div>
  <div title="o" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 0.4666666666666667;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      O
    </div>
  </div>
  <div title="r" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 0.37777777777777777;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      R
    </div>
  </div>
  <div title="k" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 0.28888888888888886;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      K
    </div>
  </div>
</div>
`;

exports[`LetterGridCard component renders exploded variant 1`] = `
<div class="sg-letter-grid-card">
  <div title="P" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 1; bottom: 25px; right: 25px;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      P
    </div>
  </div>
  <div title="a" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 0.9111111111111111; bottom: 10px; right: 10px;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      A
    </div>
  </div>
  <div title="t" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 0.8222222222222222;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      T
    </div>
  </div>
  <div title="c" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 0.7333333333333334;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      C
    </div>
  </div>
  <div title="h" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 0.6444444444444444;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      H
    </div>
  </div>
  <div title="w" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 0.5555555555555556;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      W
    </div>
  </div>
  <div title="o" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 0.4666666666666667;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      O
    </div>
  </div>
  <div title="r" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 0.37777777777777777;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      R
    </div>
  </div>
  <div title="k" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 0.28888888888888886; top: 10px; left: 10px;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      K
    </div>
  </div>
</div>
`;

exports[`LetterGridCard component renders with alternative size 1`] = `
<div class="sg-letter-grid-card">
  <div title="P" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 400px; opacity: 1;">
    <div class="sg-letter-icon__inner" style="height: 400px; width: 400px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 222.22222222222223px;">
      P
    </div>
  </div>
  <div title="a" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 400px; opacity: 0.9111111111111111;">
    <div class="sg-letter-icon__inner" style="height: 400px; width: 400px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 222.22222222222223px;">
      A
    </div>
  </div>
  <div title="t" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 400px; opacity: 0.8222222222222222;">
    <div class="sg-letter-icon__inner" style="height: 400px; width: 400px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 222.22222222222223px;">
      T
    </div>
  </div>
  <div title="c" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 400px; opacity: 0.7333333333333334;">
    <div class="sg-letter-icon__inner" style="height: 400px; width: 400px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 222.22222222222223px;">
      C
    </div>
  </div>
  <div title="h" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 400px; opacity: 0.6444444444444444;">
    <div class="sg-letter-icon__inner" style="height: 400px; width: 400px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 222.22222222222223px;">
      H
    </div>
  </div>
  <div title="w" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 400px; opacity: 0.5555555555555556;">
    <div class="sg-letter-icon__inner" style="height: 400px; width: 400px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 222.22222222222223px;">
      W
    </div>
  </div>
  <div title="o" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 400px; opacity: 0.4666666666666667;">
    <div class="sg-letter-icon__inner" style="height: 400px; width: 400px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 222.22222222222223px;">
      O
    </div>
  </div>
  <div title="r" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 400px; opacity: 0.37777777777777777;">
    <div class="sg-letter-icon__inner" style="height: 400px; width: 400px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 222.22222222222223px;">
      R
    </div>
  </div>
  <div title="k" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 400px; opacity: 0.28888888888888886;">
    <div class="sg-letter-icon__inner" style="height: 400px; width: 400px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 222.22222222222223px;">
      K
    </div>
  </div>
</div>
`;

exports[`LetterGridCard component renders with alternative word 1`] = `
<div class="sg-letter-grid-card">
  <div title="F" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 1;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      F
    </div>
  </div>
  <div title="o" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 0.8666666666666667;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      O
    </div>
  </div>
  <div title="o" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 0.7333333333333334;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      O
    </div>
  </div>
  <div title="b" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 0.6;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      B
    </div>
  </div>
  <div title="a" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 0.4666666666666667;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      A
    </div>
  </div>
  <div title="r" class="sg-letter-icon sg-letter-grid-card__letter" style="width: 80px; opacity: 0.33333333333333337;">
    <div class="sg-letter-icon__inner" style="height: 80px; width: 80px; background-color: rgb(255, 84, 61); color: rgb(255, 255, 255); font-size: 44.44444444444444px;">
      R
    </div>
  </div>
</div>
`;
